### PR TITLE
fix: disable onClick on value cells if no ouId

### PR DESCRIPTION
**Pull request by edoardo**
Right now we don't have support for period drill up/down, so in case ou
dimension is in filters, the contextual menu opens empty (peId is passed
if pe dimension is in rows/columns).

### CODE REVIEW

File: src/components/PivotTable/PivotTableValueCell.js [hunk 1]
```diff
@@ -58,10 +58,15 @@ export const PivotTableValueCell = ({
...
+    const classes = [cellContent.cellType, cellContent.valueType]
+
+    if (cellContent.ouId) {
+        classes.push('clickable')
+    }
     return (
         <PivotTableCell
             key={column}
-            classes={[cellContent.cellType, cellContent.valueType]}
+            classes={classes} //[cellContent.cellType, cellContent.valueType, isClickable]}
```
Comment (janhenrikoverland) [inline src/components/PivotTable/PivotTableValueCell.js]: I guess the comment is not needed

Comment (edoardo) [reply src/components/PivotTable/PivotTableValueCell.js]: Yep, leftover code. Cleaned up.

File: src/components/PivotTable/PivotTableValueCell.js [hunk 2]
```diff
@@ -58,13 +58,18 @@ export const PivotTableValueCell = ({
...
+    }
     return (
         <PivotTableCell
             key={column}
-            classes={[cellContent.cellType, cellContent.valueType]}
+            classes={classes}
             title={cellContent.renderedValue}
             style={style}
-            onClick={onClick}
+            onClick={onToggleContextualMenu ? onClick : undefined}
```
Comment (janhenrikoverland) [inline src/components/PivotTable/PivotTableValueCell.js]: Maybe just do `onToggleContextualMenu && onClick`

**Review by janhenrikoverland — APPROVED** [review]

File: src/components/PivotTable/PivotTableValueCell.js [hunk 3]
```diff
@@ -58,13 +58,18 @@ export const PivotTableValueCell = ({
         maxWidth: width,
     }
 
+    const classes = [cellContent.cellType, cellContent.valueType]
```
Comment (amcgee) [inline src/components/PivotTable/PivotTableValueCell.js]: Logic looks fine to me, might be a little cleaner like this:

```js
const isClickable = cellContent.cellType === CELL_TYPE_VALUE && cellContent.ouId
const onClick = () => { onToggleContextualMenu(cellRef, cellContent) }
const classes = [cellContent.cellType, cellContent.valueType, isClickable && 'clickable']

//...

return <PivotTableCell
            key={column}
            classes={classes}
            title={cellContent.renderedValue}
            style={style}
            onClick={isClickable ? onClick : undefined}
            ref={cellRef}
/>
```

**Review by amcgee — APPROVED** [review]
Minor suggestion, may or may not be cleaner

File: src/components/PivotTable/PivotTableValueCell.js (hunk 2 above)
Comment (edoardo) [reply src/components/PivotTable/PivotTableValueCell.js]: Can't do:
> index.js:1 Warning: Expected `onClick` listener to be a function, instead got `false`.

> If you used to conditionally omit it with onClick={condition && value}, pass onClick={condition ? value : undefined} instead.

File: src/components/PivotTable/PivotTableValueCell.js (hunk 3 above)
Comment (edoardo) [reply src/components/PivotTable/PivotTableValueCell.js]: Cleaned up.

File: src/components/PivotTable/PivotTableValueCell.js [hunk 4]
```diff
@@ -33,7 +27,7 @@ export const PivotTableValueCell = ({
         return (
             <PivotTableEmptyCell
                 type={cellContent?.cellType}
-                onClick={onClick}
+                onClick={onToggleContextualMenu ? onClick : undefined}
```
Comment (amcgee) [inline src/components/PivotTable/PivotTableValueCell.js:30]: @edoardo I think `onClick` isn’t defined yet here so this will break on empty cells?

Comment (edoardo) [reply src/components/PivotTable/PivotTableValueCell.js:30]: True, fixing in a new PR.

**Pull request merged**